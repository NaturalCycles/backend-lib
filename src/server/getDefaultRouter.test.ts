import { debugResource } from '../test/debug.resource'
import { expressTestService } from '../testing'

// This is too easy:
// const app = expressTestService.getGot()
// const app = getGot().extend({
//   prefixUrl: process.env.__EXPRESS_SERVER_URL__,
// })

const app = expressTestService.createApp([debugResource])
afterAll(async () => {
  await app.close()
})

test('root resource ok', async () => {
  const r = await app('').json()
  expect(r).toEqual({ ok: 1 })
})

// This started to mysteriously fail after some Got upgrade
// todo: investigate!
test.skip('should handle async error', async () => {
  const err = await app('asyncError', { throwHttpErrors: false }).json()
  expect(err).toMatchSnapshot()
})
