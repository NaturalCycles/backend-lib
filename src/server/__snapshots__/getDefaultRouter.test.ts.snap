// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle async error 1`] = `
Object {
  "data": Object {
    "httpStatusCode": 500,
  },
  "message": "debug_async_error",
}
`;
